content/post/094-js-ipfs-0.44.md

```diff
@@ -0,0 +1,177 @@
+---
+date: 2020-05-18
+url: 2020-05-18-js-ipfs-0-44
+title: js-ipfs 0.44.0 released
+header_image: js-ipfs-placeholder.png
+author: Alex Potsides
+---
+
+# 🔦 Highlights
+
+> Cancellable requests and a new browser datastore
+
+`js-IPFS@0.44.0` is out of the door with support for cancelling requests and a leaner, meaner datastore for the browser!
+
+## 🤖 Cancellable requests
+
+A user should be able to cancel long-lived asynchronous API operations. For example, if you are fetching the content for a [CID][] from the network, and that CID is not resolveable, you should be able to set a timeout value for the request, after which the content would no longer be sought and control would be returned to your code with an error message describing what happened.
+
+This is not as straightforward as it seems in JavaScript because a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) is returned from the API call to resolve a piece of content and the [Promises spec](https://promisesaplus.com/) includes nothing about cancelling a Promise.
+
+However, the browser [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) has a notion of an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) which can be used to abort web requests. They are interacted with via an [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController):
+
+```javascript
+const controller = new AbortController()
+
+setTimeout(() => {
+  controller.abort()
+}, 1000)
+
+try {
+  await result = fetch('http://example.com', {
+    signal: controller.signal
+  })
+} catch (err) {
+  console.err(err) // prints: user aborted the request
+}
+```
+
+We’ve taken this approach and integrated it into every API call, so you can now use AbortControllers to cancel requests! We’ve also used this mechanism to add a `timeout` option to every API call which internally uses an AbortSignal to stop the request after the specified amount of time:
+
+```javascript
+const cid = new CID('QmWillNeverResolve')
+
+try {
+  await ipfs.get(cid, {
+    timeout: 1000 // abort after 1000ms
+  })
+} catch (err) {
+  console.err(err)
+}
```

Comment:
Can you clarify via comment or otherwise that adding a timeout *does* throw an exception? (like in prev example)

Reply:
Done!